pkg/providers/instancetype/types.go

```diff
@@ -224,10 +224,13 @@ func awsNeurons(info *ec2.InstanceTypeInfo) *resource.Quantity {
 	count := int64(0)
 	if info.InferenceAcceleratorInfo != nil {
 		for _, accelerator := range info.InferenceAcceleratorInfo.Accelerators {
-			count += *accelerator.Count
+			// Check if the accelerator type is for Trainium instances
+			if *accelerator.Type == "neuron" {
+				count += *accelerator.Count
```

Comment:
when I query for InferenceAcceleratorInfo for `trn1` it's null in `ec2.DescribeInstanceTypes`.  Which instance type are you seeing that has this populated? 

```
> aws ec2 describe-instance-types --instance-types trn1n.32xlarge --region us-east-1
    {
        "AutoRecoverySupported": false,
        "BareMetal": false,
        "BurstablePerformanceSupported": false,
        "CurrentGeneration": true,
        "DedicatedHostsSupported": false,
        "EbsInfo": {
            "EbsOptimizedInfo": {
                "BaselineBandwidthInMbps": 80000,
                "BaselineIops": 260000,
                "BaselineThroughputInMBps": 10000,
                "MaximumBandwidthInMbps": 80000,
                "MaximumIops": 260000,
                "MaximumThroughputInMBps": 10000
            },
            "EbsOptimizedSupport": "default",
            "EncryptionSupport": "supported",
            "NvmeSupport": "required"
        },
        "FpgaInfo": null,
        "FreeTierEligible": false,
        "GpuInfo": null,
        "HibernationSupported": false,
        "Hypervisor": "nitro",
        "InferenceAcceleratorInfo": null,
        "InstanceStorageInfo": {
            "Disks": [
                {
                    "Count": 4,
                    "SizeInGB": 1900,
                    "Type": "ssd"
                }
            ],
            "EncryptionSupport": "required",
            "NvmeSupport": "required",
            "TotalSizeInGB": 7600
        },
        "InstanceStorageSupported": true,
        "InstanceType": "trn1n.32xlarge",
...
```

Reply:
Based on the provided output of the aws ec2 describe-instance-types command, the InferenceAcceleratorInfo field is indeed null for the trn1n.32xlarge instance type. This indicates that the trn1n.32xlarge instance type does not have inference accelerators associated with it.

Further the InferenceAcceleratorInfo field being null means that there are no specific inference accelerators available for the trn1n.32xlarge instance type.

Reply:
Correct me if I'm wrong

Reply:
I don't think that is correct though. There should be accelerators for these. If DescribeInstanceTypes is not showing it, then the code in this PR won't work properly. Were you able to find other instance types that have accelerator info where this code will work?

Reply:
Yes you are right upon further reviewing I noticed that I've misunderstood 
Can you give me a push where the change needs to be done ?
I'll get back to you if I have something.

Reply:
I think we can just hardcode the accelerators until it's fixed in DescribeInstanceTypes. You can get the data here: https://aws.amazon.com/ec2/instance-types/